ports/esp32/modmachine.c

```diff
@@ -94,41 +94,49 @@ STATIC bool is_soft_reset = 0;
 int esp_clk_cpu_freq(void);
 #endif
 
-STATIC mp_obj_t mp_machine_get_freq(void) {
-    return mp_obj_new_int(esp_rom_get_cpu_ticks_per_us() * 1000000);
-}
-
-STATIC void mp_machine_set_freq(size_t n_args, const mp_obj_t *args) {
-    mp_int_t freq = mp_obj_get_int(args[0]) / 1000000;
-    if (freq != 20 && freq != 40 && freq != 80 && freq != 160
-        #if !CONFIG_IDF_TARGET_ESP32C3
-        && freq != 240
-        #endif
-        ) {
-        #if CONFIG_IDF_TARGET_ESP32C3
-        mp_raise_ValueError(MP_ERROR_TEXT("frequency must be 20MHz, 40MHz, 80Mhz or 160MHz"));
+STATIC mp_obj_t machine_freq(size_t n_args, const mp_obj_t *args) {
```

Comment:
This needs to be two functions: `mp_machine_set_freq()`, and `mp_machine_get_freq()`.  The CI is failing at the moment because if this error.